tests/monitoring_test.go

```diff
@@ -156,15 +161,158 @@ var _ = Describe("Prometheus Alerts", func() {
+					},
+					PersistentVolumeSource: core.PersistentVolumeSource{
+						CSI: &core.CSIPersistentVolumeSource{
+							Driver: "openshift-storage.rbd.csi.ceph.com",
+							VolumeAttributes: map[string]string{
+								"clusterID":     "test-cluster",
+								"mounter":       "rbd",
+								"imageFeatures": "layering,deep-flatten,exclusive-lock",
+								"mapOptions":    mapOptions,
+							},
+							VolumeHandle: vmName,
+						},
+					},
+				},
+			}
+			Expect(apiClient.Create(ctx, pv)).ToNot(HaveOccurred())
+
+			volumeMode := core.PersistentVolumeBlock
+			pvc = &core.PersistentVolumeClaim{
+				ObjectMeta: metav1.ObjectMeta{
+					Name:      vmName,
+					Namespace: strategy.GetNamespace(),
+				},
+				Spec: core.PersistentVolumeClaimSpec{
+					VolumeName: vmName,
+					VolumeMode: &volumeMode,
+					AccessModes: []core.PersistentVolumeAccessMode{
+						core.ReadWriteOnce,
+					},
+					Resources: core.ResourceRequirements{
+						Requests: core.ResourceList{
+							core.ResourceStorage: resource.MustParse("1Gi"),
+						},
+					},
+				},
+			}
+			Expect(apiClient.Create(ctx, pvc)).ToNot(HaveOccurred())
+		}
+
+		var createResources = func(createDataVolume bool, rxbounceEnabled bool) string {
+			vmName := fmt.Sprintf("testvmi-%v", rand.String(10))
+
+			var volumes []kubevirtv1.Volume
+
+			if createDataVolume {
+				createPVCAndPV(vmName, rxbounceEnabled)
+				volumes = append(volumes, kubevirtv1.Volume{
+					Name: vmName,
+					VolumeSource: kubevirtv1.VolumeSource{
+						DataVolume: &kubevirtv1.DataVolumeSource{
+							Name: vmName,
+						},
+					},
+				})
+			}
+
+			vmi := NewMinimalVMIWithNS(strategy.GetNamespace(), vmName)
+			vmi.Spec = kubevirtv1.VirtualMachineInstanceSpec{
+				Volumes: volumes,
+			}
+			vm = NewVirtualMachine(vmi)
+			vm.Spec.Running = pointer.Bool(false)
+			eventuallyCreateVm(vm)
+
+			return vmName
+		}
+
+		It("Should not create kubevirt_ssp_vm_rbd_volume when not using DataVolume or PVC", func() {
+			vmName := createResources(false, true)
+			seriesShouldNotBeDetected(fmt.Sprintf("kubevirt_ssp_vm_rbd_volume{name='%s'}", vmName))
+			alertShouldNotBeActive("VirtualMachineCRCErrors")
+		})
+
+		It("Should not fire VirtualMachineCRCErrors when rxbounce is enabled", func() {
+			vmName := createResources(true, true)
+			waitForSeriesToBeDetected(fmt.Sprintf("kubevirt_ssp_vm_rbd_volume{name='%s', rxbounce_enabled='true'}", vmName))
+			alertShouldNotBeActive("VirtualMachineCRCErrors")
+		})
+
+		It("Should fire VirtualMachineCRCErrors is disabled", func() {
+			vmName := createResources(true, false)
+			waitForSeriesToBeDetected(fmt.Sprintf("kubevirt_ssp_vm_rbd_volume{name='%s', rxbounce_enabled='false'}", vmName))
+			waitForAlertToActivate("VirtualMachineCRCErrors")
+		})
+	})
 })
 
+func checkAlert(alertName string) *promApiv1.Alert {
+	alerts, err := getPrometheusClient().Alerts(context.TODO())
+	Expect(err).ShouldNot(HaveOccurred())
```

Comment:
Can you return the error instead of using `Expect()`? This function is called only inside `Eventually()` so using `Expect()` would break out of the `Eventually()` loop and it would not be called again.

Reply:
updated